packages/jest-environment-puppeteer/src/PuppeteerEnvironment.js

```diff
@@ -32,7 +32,12 @@ class PuppeteerEnvironment extends NodeEnvironment {
     const config = await readConfig()
     this.global.puppeteerConfig = config
 
-    const wsEndpoint = fs.readFileSync(WS_ENDPOINT_PATH, 'utf8')
+    let wsEndpoint;
+    try {
+        wsEndpoint = fs.default.readFileSync(WS_ENDPOINT_PATH + ((process.ppid) ? process.pid : ""), 'utf8')
+    } catch (e) {
+        wsEndpoint = fs.default.readFileSync(WS_ENDPOINT_PATH + process.ppid, 'utf8')
+    }
```

Comment:
I don't understand the logic here. If there is a ppid, then we use pid but if there is an error then we fallback using ppid??

Reply:
The correct logic should be to fallback to the using the old name without the pid added if ppid is not defined/supported. Then attempt to read file using pid and if that fails ppid. I see no reason why do it in a specific order, I was trying to use the same pattern as in the global,js.